// syscoin/lib/commands.js
module.exports = {
  /* == Blockchain == */
  getBestBlockHash: "getbestblockhash",
  getBlock: "getblock",
  getBlockchainInfo: "getblockchaininfo",
  getBlockCount: "getblockcount",
  getBlockHash: "getblockhash",
  getBlockHeader: "getblockheader",
  getChainTips: "getchaintips",
  getDifficulty: "getdifficulty",
  getMemPoolAncestors: "getmempoolancestors",
  getMemPoolDescendants: "getmempooldescendants",
  getMemPoolEntry: "getmempoolentry",
  getMemPoolInfo: "getmempoolinfo",
  getRawMemPool: "getrawmempool",
  getTxOut: "gettxout",
  getTxOutProof: "gettxoutproof",
  getTxOutSetInfo: "gettxoutsetinfo",
  verifyChain: "verifychain",
  verifyTxOutProof: "verifytxoutproof",

  /* == Control == */
  getInfo: "getinfo",
  help: "help",
  stop: "stop",

  /* == Generating == */
  generate: "generate",
  generateToAddress: "generatetoaddress",

  /* == Mining == */
  getAuxBlock: "getauxblock",
  getBlockTemplate: "getblocktemplate",
  getMiningInfo: "getmininginfo",
  getNetworkHashPs: "getnetworkhashps",
  prioritiseTransaction: "prioritisetransaction",
  submitBlock: "submitblock",

  /* == Network == */
  addNode: "addnode",
  clearBanned: "clearbanned",
  disconnectNode: "disconnectnode",
  getAddedNodeInfo: "getaddednodeinfo",
  getConnectionCount: "getconnectioncount",
  getNetTotals: "getnettotals",
  getNetworkInfo: "getnetworkinfo",
  getPeerInfo: "getpeerinfo",
  listBanned: "listbanned",
  ping: "ping",
  setBan: "setban",

  /*  == Rawtransactions == */
  createRawTransaction: "createrawtransaction",
  decodeRawTransaction: "decoderawtransaction",
  decodeScript: "decodescript",
  fundRawTransaction: "fundrawtransaction",
  getRawTransaction: "getrawtransaction",
  sendRawTransaction: "sendrawtransaction",
  signRawTransaction: "signrawtransaction",

  /* == Util == */
  createMultiSig: "createmultisig",
  estimateFee: "estimatefee",
  estimatePriority: "estimatepriority",
  estimateSmartFee: "estimatesmartfee",
  estimateSmartPriority: "estimatesmartpriority",
  signMessageWithPrivKey: "signmessagewithprivkey",
  validateAddress: "validateaddress",
  verifyMessage: "verifymessage",
  walletLock: "walletlock",
  walletPassphrase: "walletpassphrase",
  walletPassphraseChange: "walletpassphrasechange",

  /* == Wallet == */
  abandonTransaction: "abandontransaction",
  addMultiSigAddress: "addmultisigaddress",
  addWitnessAddress: "addwitnessaddress",
  aliasAffiliates: "aliasaffiliates",
  aliasAuthenticate: "aliasauthentice",
  aliasBalance: "aliasbalance",
  aliasCount: "aliascount",
  aliasFilter: "aliasfilter",
  aliasHistory: "aliashistory",
  aliasInfo: "aliasinfo",
  aliasList: "aliaslist",
  aliasNew: "aliasnew",
  aliasPay: "aliaspay",
  aliasUpdate: "aliasupdate",
  backupWallet: "backupwallet",
  certCount: "certcount",
  certFilter: "certfilter",
  certHistory: "certhistory",
  certInfo: "certinfo",
  certList: "certlist",
  certNew: "certnew",
  certTransfer: "certtransfer",
  certUpdate: "certupdate",
  dumpPrivKey: "dumpprivkey",
  dumpWallet: "dumpwallet",
  encryptWallet: "encryptwallet",
  escrowAcknowledge: "escrowacknowledge",
  escrowClaimRefund: "escrowclaimrefund",
  escrowClaimRelease: "escrowclaimrelease",
  escrowCompleteRefund: "escrowcompleterefund",
  escrowCompleteRelease: "escrowcompleterelease",
  escrowCount: "escrowcount",
  escrowFeedback: "escrowfeedback",
  escrowFilter: "escrowfilter",
  escrowHistory: "escrowhistory",
  escrowInfo: "escrowinfo",
  escrowList: "escrowlist",
  escrowNew: "escrownew",
  escrowRefund: "escrowrefund",
  escrowRelease: "escrowrelease",
  generateEscrowMultisig: "generateescrowmultisig",
  getAccount: "getaccount",
  getAccountAddress: "getaccountaddress",
  getAddressesByAccount: "getaddressesbyaccount",
  getBalance: "getbalance",
  getNewAddress: "getnewaddress",
  getRawChangeAddress: "getrawchangeaddress",
  getReceivedByAccount: "getreceivedbyaccount",
  getReceivedByAddress: "getreceivedbyaddress",
  getTransaction: "gettransaction",
  getUnconfirmedBalance: "getunconfirmedbalance",
  getV2Address: "getv2address",
  getWalletInfo: "getwalletinfo",
  getZAddress: "getzaddress",
  importAddress: "importaddress",
  importPrivKey: "importprivkey",
  importPrunedFunds: "importprunedfunds",
  importPubKey: "importpubkey",
  importWallet: "importwallet",
  keypoolRefill: "keypoolrefill",
  listAccounts: "listaccounts",
  listAddressGroupings: "listaddressgroupings",
  listLockUnspent: "listlockunspent",
  listReceivedByAccount: "listreceivedbyaccount",
  listReceivedByAddress: "listreceivedbyaddress",
  listSinceBlock: "listsinceblock",
  listTransactions: "listtransactions",
  listUnspent: "listunspent",
  lockUnspent: "lockunspent",
  messageInfo: "messageinfo",
  messageNew: "messagenew",
  messageReceiveCount: "messagereceivecount",
  messageReceiveList: "messagereceivelist",
  messageSentCount: "messagesentcount",
  messageSentList: "messagesentlist",
  move: "move",
  offerAccept: "offeraccept",
  offerAcceptAcknowledge: "offeracceptacknowledge",
  offerAcceptCount: "offeracceptcount",
  offerAcceptFeedback: "offeracceptfeedback",
  offerAcceptList: "offeracceptlist",
  offerAddWhitelist: "offeraddwhitelist",
  offerClearWhitelist: "offerclearwhitelist",
  offerCount: "offercount",
  offerFilter: "offerfilter",
  offerHistory: "offerhistory",
  offerInfo: "offerinfo",
  offerLink: "offerlink",
  offerList: "offerlist",
  offerNew: "offernew",
  offerRemoveWhitelist: "offerremovewhitelist",
  offerUpdate: "offerupdate",
  offerWhitelist: "offerwhitelist",
  removePrunedFunds: "removeprunedfunds",
  sendFrom: "sendfrom",
  sendMany: "sendmany",
  sendToAddress: "sendtoaddress",
  setAccount: "setaccount",
  setTxFee: "settxfee",
  signMessage: "signmessage",
  syscoinDecodeRawTransaction: "syscoindecoderawtransaction",
  syscoinSignRawTransaction: "syscoinsignrawtransaction"
};
